ipn/ipnlocal/c2n.go

```diff
@@ -77,3 +88,92 @@ func (b *LocalBackend) handleC2N(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "unknown c2n path", http.StatusBadRequest)
 	}
 }
+
+func (b *LocalBackend) handleC2NUpdate(w http.ResponseWriter, r *http.Request) {
+	type updateResponse struct {
+		Err       string // error message, if any
+		Enabled   bool   // user has opted-in to remote updates
+		Supported bool   // Tailscale supports updating this OS/platform
+		Started   bool
+	}
+	var res updateResponse
+	res.Enabled = envknob.AllowsRemoteUpdate()
+	res.Supported = runtime.GOOS == "windows" || (runtime.GOOS == "linux" && distro.Get() == distro.Debian)
+
+	switch r.Method {
+	case "GET", "POST":
+	default:
+		http.Error(w, "bad method", http.StatusMethodNotAllowed)
+		return
+	}
+
+	defer func() {
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(res)
+	}()
+
+	if r.Method == "GET" {
+		return
+	}
+
+	if !res.Enabled {
+		res.Err = "not enabled"
+		return
+	}
+
+	if !res.Supported {
+		res.Err = "not supported"
+		return
+	}
+	cmdTS, err := findCmdTailscale()
+	if err != nil {
+		res.Err = fmt.Sprintf("failed to find cmd/tailscale binary: %v", err)
+		return
+	}
+	var ver struct {
+		Long string `json:"long"`
+	}
+	out, err := exec.Command(cmdTS, "version", "--json").Output()
+	if err != nil {
+		res.Err = fmt.Sprintf("failed to find cmd/tailscale binary: %v", err)
+		return
+	}
+	if err := json.Unmarshal(out, &ver); err != nil {
+		res.Err = "invalid JSON from cmd/tailscale version --json"
+		return
+	}
+	if ver.Long != version.Long {
+		res.Err = "cmd/tailscale version mismatch"
+		return
+	}
+	cmd := exec.Command(cmdTS, "update", "--yes")
+	if err := cmd.Start(); err != nil {
+		res.Err = fmt.Sprintf("failed to start cmd/tailscale update: %v", err)
+		return
+	}
+	res.Started = true
+	defer cmd.Wait()
```

Comment:
There might be a race condition here on Windows:
1. We start the update process
2. `tailscale.exe` copies itself and kicks off the update process
3. `msiexec` stops this process during the update before the `selfCopy` exits(?)
4. This doesn't return because the process is dead

This seems fairly unlikely, but worth flagging?